### PR TITLE
Fix: undefined method downcase for nil

### PR DESCRIPTION
**Pull request by BryanEliDimas**
Fixes [this error](https://sentry.cnx.org/openstax/accounts/issues/126859/) found by Sentry.

Solution: do what the old flow did — instead of `by_verified_email`, use `LookupUsers.by_email_or_username` which is smarter and handles the cases when the email from social provider is `nil` but also, it looks up users by username 🤷‍♂.

### CODE REVIEW

File: spec/handlers/newflow/oauth_callback_spec.rb [hunk 1]
```diff
@@ -73,5 +73,18 @@ module Newflow
         end
       end
     end
+
+    context 'when authentication found, but the response from social provider does not include email' do
+      let(:request) {
+        create_newflow_user(email)
```
Comment (steveburkett) [inline spec/handlers/newflow/oauth_callback_spec.rb]: these seem like things that should be split into lets and before do/end blocks?  the info is a let, create_newflow_user is a before, the request is the subject (object being tested)?  separating like this creates clarity, and clarity improves readability and maintainability ... 

http://www.betterspecs.org/#subject

File: spec/handlers/newflow/oauth_callback_spec.rb [hunk 2]
```diff
@@ -73,5 +73,18 @@ module Newflow
...
+        info = { email: nil, name: Faker::Name.name }
+        MockOmniauthRequest.new 'facebook', Faker::Internet.uuid, info
+      }
+
+      it 'doesn\'t blow up' do
```
Comment (steveburkett) [inline spec/handlers/newflow/oauth_callback_spec.rb]: how about it 'processes the omniauth request without error'?  doesnt blow up doesnt add much information to the it block when you look at the rspec test output.

Comment (BryanEliDimas) [reply spec/handlers/newflow/oauth_callback_spec.rb:84]: Thanks for the suggestion. Fixed.

File: spec/handlers/newflow/oauth_callback_spec.rb (hunk 1 above)
Comment (BryanEliDimas) [reply spec/handlers/newflow/oauth_callback_spec.rb:79]: Will do. Great suggestions! 🙏

Comment (BryanEliDimas) [reply spec/handlers/newflow/oauth_callback_spec.rb:102]: Done.

**Pull request merged**